packages/github-actions/teams-toolkit-cicd/src/buildMapQuerier.ts

```diff
@@ -0,0 +1,47 @@
+import {Commands} from './constant'
+import {InternalError} from './errors'
+import {buildMap} from './buildMap'
+
+export class BuildMapQuerier {
+  static instance: BuildMapQuerier
+  static validOutputs: string[] = [Commands.NpmInstall, Commands.NpmRunBuild]
+
+  private constructor() {}
+  static async getInstance(): Promise<BuildMapQuerier> {
+    if (!BuildMapQuerier.instance) {
+      BuildMapQuerier.instance = new BuildMapQuerier()
+    }
+
+    return BuildMapQuerier.instance
+  }
+
+  query(cap: string, lang?: string): string[] {
+    const commands = this._query(cap, lang)
+
+    if (commands.some(value => !BuildMapQuerier.validOutputs.includes(value))) {
```

Comment:
Nit: This check looks useless, we don't need to verify the string in our code. Redundant would involve inconsistent trouble.

Reply:
This is for the case when developers modify the buildMap.json. It's also legacy code, will be deleted.